File: backend/app/schemas/initiative.py
"""Pydantic schemas for Key Business Initiatives."""

from typing import List, Optional

from pydantic import BaseModel


class InitiativeProjectSummary(BaseModel):
    """Lightweight project summary included in initiative detail responses."""
    id: str
    name: str
    health: Optional[str] = None
    pct_complete: float = 0.0
    priority: Optional[str] = None
    planned_it_start: Optional[str] = None


class InitiativeOut(BaseModel):
    id: str
    name: str
    description: Optional[str] = None
    sponsor: Optional[str] = None
    status: str = "Active"
    it_involvement: bool = False
    priority: Optional[str] = None
    target_start: Optional[str] = None
    target_end: Optional[str] = None
    sort_order: int = 0
    created_at: Optional[str] = None
    updated_at: Optional[str] = None
    project_count: int = 0
    projects: List[InitiativeProjectSummary] = []

    @classmethod
    def from_row(cls, row: dict) -> "InitiativeOut":
        projects = [
            InitiativeProjectSummary(**p) for p in row.get("projects", [])
        ]
        return cls(
            id=row["id"],
            name=row["name"],
            description=row.get("description"),
            sponsor=row.get("sponsor"),
            status=row.get("status", "Active"),
            it_involvement=bool(row.get("it_involvement", 0)),
            priority=row.get("priority"),
            target_start=row.get("target_start"),
            target_end=row.get("target_end"),
            sort_order=row.get("sort_order", 0) or 0,
            created_at=row.get("created_at"),
            updated_at=row.get("updated_at"),
            project_count=row.get("project_count", len(projects)),
            projects=projects,
        )

File: frontend/src/types/initiative.ts
export interface InitiativeProjectSummary {
  id: string;
  name: string;
  health: string | null;
  pct_complete: number;
  priority: string | null;
  planned_it_start: string | null;
}

export interface Initiative {
  id: string;
  name: string;
  description: string | null;
  sponsor: string | null;
  status: "Active" | "Complete" | "On Hold";
  it_involvement: boolean;
  priority: string | null;
  target_start: string | null;
  target_end: string | null;
  sort_order: number;
  created_at: string | null;
  updated_at: string | null;
  project_count: number;
  projects: InitiativeProjectSummary[];
}
